boa_engine/src/builtins/string/mod.rs

```diff
@@ -2140,10 +2140,10 @@ pub(crate) fn get_substitution(
     while let Some(first) = chars.next() {
         if first == '$' {
             let second = chars.next();
-            let second_is_digit = second.map_or(false, |ch| ch.is_digit(10));
+            let second_is_digit = second.map_or(false, |ch| ch.is_ascii_digit());
```

Comment:
I feel like this is a bit more clear, since it avoids the redundant closure:
```suggestion
            let second_is_digit = second.as_ref().map_or(false, char::is_ascii_digit);
```